### DISCUSSION TITLE
Working with long content

### DISCUSSION THREAD

**Discussion by arminta7** [nhaouari/obsidian-textgenerator-plugin#41] [Q&A]
Hello!

Readwise Reader has a neat way of working with long texts. In their prompt template they include:

"""

{% if (document.content | count_tokens) > 1000 %}

   {{ document.content | central_sentences | join('\n\n') }}

{% else %}

   {{ document.content }}

{% endif %}

"""

What would be the equivalent way of doing this with the text generator plugin? Either via text or template?

**Comment by nhaouari** [comment]
I'm not a Readwise user, but from what I can tell, is their template designed to display only the central sentences of a text if it is longer than 1000 tokens.

**Reply by arminta7** [reply]
Correct. Then you would use that to apply your prompt to. 

For example:

```
Summarize the document below entitled "{{ document.title }}":

"""

{% if (document.content | count_tokens) > 1000 %}

   {{ document.content | central_sentences | join('\n\n') }}

{% else %}

   {{ document.content }}

{% endif %}

"""
```

This would allow you to summarize a very long document by staying within the token limit.

**Comment by nhaouari** [comment]
A helper function is added to the Text Generator Template engine that allows you to take a substring of text.

To use this function, simply use the following syntax:

{{{substring var startPosition endPosition}}}

For example, if you only want the first 100 characters of the selected text, you would use:

{{{substring selection 0 99}}}
